src/hex/coordinate.rs

```diff
@@ -36,9 +36,161 @@ macro_rules! axial {
 }
 pub use axial;
 
+/// Vertex spin is a orientation of the vertex.
+///
+/// A vertex needs to know its `spin`. Spin correlates to which side [`VertexSpin::Up`] or [`VertexSpin::Down`]
+/// has two hexagons.
+///
+/// see [`Vertex`]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(PartialEq, Eq, Copy, Clone, Hash, Debug)]
+pub enum VertexSpin {
+    Up,
+    Down,
+}
+
+/// A vertex direction denotes the direction from the hexagon center the vertex is.
+///
+/// Reference pointy-top hexagons for vertex direction, where up being directly above the center.
+///
+/// see [`Vertex`]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+pub enum VertexDirection {
```

Comment:
Not that I can think of any implementation direction at this point, but is there any value drawing parallels between this and HexDirection, given the relationship between these?

Reply:
True, both HexDirection and VertexDirection are a 6-direction unit. However, I think keeping them separate in terms of clarity is beneficial. This is because they refer to two separate orientations of a hexagon. Looking at the functions inside Axial, the shared verts, HexDirection is used for sure but more specifically to find an edge not the vertex.

Reply:
Sounds good to me.